packages/flutter/lib/src/widgets/localized_resources.dart

```diff
@@ -0,0 +1,354 @@
+/// widgets that have created a dependency by calling
+/// `LocalizedResources.of(context)` will be rebuilt after the resources
+/// for the new locale have been loaded.
+///
+/// ## Sample code
+///
+/// This following class is defined in terms of the
+/// [Dart `intl` package](https://github.com/dart-lang/intl). Using the `intl`
+/// package isn't required.
+///
+/// ```dart
+/// class MyLocalizedResources {
+///   MyLocalizedResources(this.locale);
+///
+///   final Locale locale;
+///
+///   static Future<MyLocalizedResources> load(Locale locale) {
+///     return initializeMessages(locale.toString())
+///       .then((Null _) {
+///         return new Future<MyLocalizedResources>.value(new MyLocalizedResources(locale));
+///       });
+///   }
+///
+///   static MyLocalizedResources of(BuildContext context) {
+///     return LocalizedResources.of(context).resourcesFor<MyLocalizedResources>(MyLocalizedResources);
+///   }
+///
+///   String title() => Intl.message('<title>', name: 'title', locale: locale.toString());
+///   // ... more Intl.message() methods like title()
+/// }
+/// ```
+/// A class based on the `intl` package imports a generated message catalog that provides
+/// the `initializeMessages()` function and the per-locale backing store for `Intl.message()`.
+/// The message catalog is produced by an `intl` tool that analyzes the source code for
+/// classes that contain `Intl.message()` calls. In this case that would just be the
+/// `MyLocalizedResources` class.
+///
+/// One could choose another approach for loading localized resources and looking them up while
+/// still conforming to the structure of this example.
+class LocalizedResources extends StatefulWidget {
+  LocalizedResources({
+    Key key,
+    @required this.locale,
+    this.delegate,
+    this.child
+  }) : super(key: key) {
+    assert(locale != null);
+  }
+
+  /// The resources returned by [LocalizedResources.of] will be specific to this locale.
+  final Locale locale;
+
+  /// This delegate is responsible for loading and caching resources for `locale`.
+  ///
+  /// Typically defined with [DefaultLocalizedResourcesDelegate].
+  final LocalizedResourcesDelegate delegate;
+
+  /// The widget below this widget in the tree.
+  final Widget child;
+
+  /// Returns the localized resources for the widget tree that corresponds to
+  /// [BuildContext] `context`.
+  ///
+  /// This method is typically combined with [LocalizedResourcesData.for] to defined
+  /// a static method that looks up a collection of related related resources. For
+  /// example Flutter looks up Material resources with a method defined like this:
+  ///
+  /// ```dart
+  /// static LocalizedMaterialResources of(BuildContext context) {
+  ///  return LocalizedResources.of(context)
+  ///    .resourcesFor<LocalizedMaterialResources>(LocalizedMaterialResources);
+  /// }
+  /// ```
+  static LocalizedResourcesData of(BuildContext context) {
```

Comment:
instead of having the LocalizedResourcesData middleman, I would just have two static methods, like `Strings.locale(context)` and `Strings.obtain(context, Foo)` or some such.